easybuild/easyconfigs/c/Caffe/Caffe-1.0-intel-2017b-CUDA-9.1.85-Python-2.7.14.eb

```diff
@@ -0,0 +1,68 @@
+# -*- mode: python; -*-
+# EasyBuild reciPY for Caffe as per https://github.com/easybuilders/easybuild
+#
+# Copyright:: Copyright 2017 UL HPC
+# Authors::   UL HPC Team <hpc-sysadmins@uni.lu>
+# License::   MIT/GPL
+#
+easyblock = 'CMakeMake'
+
+name = 'Caffe'
+version = '1.0'
+
+versionsuffix = '-Python-%(pyver)s'
+cudaversion = '9.1.85'
+versionsuffix = '-CUDA-%s-Python-%%(pyver)s' % cudaversion
+
+homepage = 'https://github.com/BVLC/caffe'
+description = """
+Caffe is a deep learning framework made with expression, speed,
+and modularity in mind. It is developed by the Berkeley Vision
+and Learning Center (BVLC) and community contributors.
+"""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/BVLC/caffe/archive/']
+sources = ['%(version)s.tar.gz']
+checksums = ['71d3c9eb8a183150f965a465824d01fe82826c22505f7aa314f700ace03fa77f']
+
+builddependencies = [
+    ('CMake', '3.9.1'),
+]
+
+# See http://caffe.berkeleyvision.org/installation.html
+dependencies = [
+    ('Python', '2.7.14'),
+    ('Boost', '1.66.0', '-Python-%(pyver)s'),
+    ('protobuf', '3.4.0'),
+    ('protobuf-python', '3.4.0', '-Python-%(pyver)s'),
+    ('glog', '0.3.5'),
+    ('gflags', '2.2.1'),
+    ('HDF5', '1.8.20'),
+    ('LMDB', '0.9.21'),
+    ('LevelDB', '1.18'),
+    ('snappy', '1.1.7'),
+    ('scikit-image', '0.13.1', '-Python-%(pyver)s'),
+    ('cuDNN', '7.0.5', '-CUDA-' + cudaversion, True),
+]
+
+configopts = (
+    '-DBLAS=mkl -DUSE_OPENCV=0 -DUSE_MKL2017_AS_DEFAULT_ENGINE=1 -DCPU_ONLY=0'
+    '-DUSE_CUDNN=1 -DCUDA_ARCH_BIN="35 60" -DCUDA_ARCH_NAME=Manual'
```

Comment:
I assume the `CUDA_ARCH_BIN` option indicates for which CUDA compute capabilities `Caffe` should be compiled?

We should make this as extensive as possible, see also the `TensorFlow` easyconfig for `GPU` we merged recently: https://github.com/easybuilders/easybuild-easyconfigs/pull/5769/files#diff-078e3b96d3f630f14cc4fad1565bd449R34

Reply:
The issue is that by default is also tries compute capabilities 2.0 which fails cause Cuda 9 does no longer support them.

So either we give here a full list of 3.0 and up or we patch the `CMakeLists.txt` to not use cc 2.0 by default.

Reply:
What is the full list of compute capabilities it targets?

If that looks fairly complete, I guess patching out `2.0` makes most sense.

If it's a limited list, or doesn't include very recent architectures, I would go for specifying the list that @akesandgren included in the `TensorFlow` `goolfc` easyconfig?

Reply:
Maybe this should become an EasyBuild configuration option though, which list of compute capabilities to build for, which can then be used as a constant in easyconfigs?

Reply:
from the cmake file:
# Known NVIDIA GPU achitectures Caffe can be compiled for.
# This list will be used for CUDA_ARCH_NAME = All option
set(Caffe_known_gpu_archs "20 21(20) 30 35 50 60 61")

I think we should stay on the safe side and only use "30 35 50 60 61".